src/tally.F90

```diff
@@ -2417,10 +2419,29 @@ subroutine score_surface_current(p)
               t % results(1, filter_index) % value = &
                    t % results(1, filter_index) % value + p % wgt
             end if
+
+            ! IN_BOTTOM
+            if (ijk0(1) >= 1 .and. ijk0(2) >= 1 .and. ijk0(3) >= 0 .and. &
+                 ijk0(1) <= m % dimension(1) .and. &
+                 ijk0(2) <= m % dimension(2) .and. &
+                 ijk0(3) <  m % dimension(3)) then
```

Comment:
I agree. There are a lot of nested loops and repeated code in here. I'll make this change and see if there is any way to make a loop over dimensions.